arch/arc/core/regular_irq.S

```diff
@@ -178,6 +177,11 @@ _rirq_common_interrupt_swap:
 .balign 4
 _rirq_return_from_coop:
 
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
+	lr r0, [_ARC_V2_SEC_STAT]
+	bset r0, r0, _ARC_V2_SEC_STAT_IRM_BIT
```

Comment:
not all interrupts are secure, or return to secure mode - why set IRM?

Reply:
here, rtie is used to do a dispatch.  The interruptted context could be in normal mode and the target thread must be in secure mode. So the  IRM must be set to the correct value.

Reply:
Why must target thread be in secure mode?